src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -279,25 +279,31 @@ def _build_service_principal(client, name, url, client_secret):
     print('done')
     return service_principal
 
-def _add_role_assignment(role, service_principal):
+def _add_role_assignment(role, service_principal, delay=2, output=True):
     # AAD can have delays in propagating data, so sleep and retry
-    sys.stdout.write('waiting for AAD role to propagate.')
+    if output:
+        sys.stdout.write('waiting for AAD role to propagate.')
+    success = False
     for x in range(0, 10):
         from azure.cli.command_modules.role.custom import create_role_assignment
         try:
             # TODO: break this out into a shared utility library
             create_role_assignment(role, service_principal)
+            success = True
             break
         except CloudError as ex:
             if ex.message == 'The role assignment already exists.':
+                success = True
```

Comment:
This is actually a good case of using `for ... else ...` control flow. https://docs.python.org/2/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops